src/java/picard/sam/util/ReadNameParsingUtils.java

```diff
@@ -29,24 +29,29 @@
 public class ReadNameParsingUtils {
 
     /**
-     * Single pass method to parse the read name for the default regex.  This will only insert the 2nd to the 4th
-     * tokens (inclusive).  It will also stop after the fifth token has been successfully parsed.
+     * Single pass method to parse the read name for the default regex.  Examines the last three fields as split by the delimiter.
      */
     public static int getRapidDefaultReadNameRegexSplit(final String readName, final char delim, final int[] tokens) {
         int tokensIdx = 0;
         int prevIdx = 0;
+        int numFields = 1;
+        for (int i = 0; i < readName.length(); i++) {
```

Comment:
This is all getting very complicated.  I wonder how much slower it would be to use either the non-regex `StringUtil.split(readName, fieldArray, ":")` and then use `rapidParseToInt` on the last three fields.